src/com/spikes2212/utils/TalonSRXEncoder.java

```diff
@@ -0,0 +1,81 @@
+package com.spikes2212.utils;
+
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.PIDSourceType;
+import edu.wpi.first.wpilibj.SendableBase;
+import edu.wpi.first.wpilibj.smartdashboard.SendableBuilder;
+
+public class TalonSRXEncoder extends SendableBase implements PIDSource{
+	
+	/**
+	 * This class makes an encoder that is connected to a {@link WPI_TalonSRX} to a
+	 * PIDSource.
+	 *
+	 * @author Tuval
+	 */
+	private TalonSRX talon;
+	private PIDSourceType type;
+	private int countsPerRevolution;
```

Comment:
change `countsPerRevolution` to `distancePerPulse` and implement it in a fashion similar to the way it is implemented in `Encoder`